hg/utils.py

```diff
@@ -69,3 +73,37 @@ def copy_unique(model: ModelT) -> ModelT:
     if hasattr(copy, "uid"):
         setattr(copy, "uid", uid())
     return copy
+
+
+ViewInterval = Tuple[int, int]
+GenomicRegion = Tuple[str, int, int]
+
+
+def overlap(a: ViewInterval, b: ViewInterval) -> bool:
+    return a[1] >= b[0] and a[0] <= b[1]
+
+
+class Scale:
+    def __init__(self, chromsizes: Iterable[Tuple[str, int]]) -> None:
+        offsets = itertools.accumulate(map(itemgetter(1), chromsizes), initial=0)
+        self._offsets = OrderedDict(
+            (name, (size, offset)) for (name, size), offset in zip(chromsizes, offsets)
+        )
+
+    def __call__(self, region: GenomicRegion) -> ViewInterval:
```

Comment:
Should be `List[GenomicRegion]` -> `ViewInterval`, no?

I think we were thinking about allowing a scalar region as a special case, but could be difficult if you wanted to allow duck typing on the container (list vs tuple). Maybe it's easier to enforce it always being a list of regions.

A list of regions also needs a contiguity check. Q: Should the input intervals be expected to be in order?